### ISSUE TITLE
Mistake in vector leaf train

### ISSUE THREAD

**Issue by oOTWK** [dmlc/xgboost#9960]
There is a logic mistake in the vector leaf training at `ApplyTreeSplit` https://github.com/dmlc/xgboost/blob/master/src/tree/hist/evaluate_splits.h#L705.

The error is in the gain calculation. The current implementation looks like the following.

```
CalcWeight(*param_, left_sum, param_->learning_rate, left_weight);
CalcWeight(*param_, right_sum, param_->learning_rate, right_weight);
gain_[left_child] = CalcGainGivenWeight(*param_, left_sum, left_weight);
gain_[right_child] = CalcGainGivenWeight(*param_, right_sum, right_weight);
```

Weight is calculated as base_weight * learning_rate and that weight is used to calculate the gain. The learning rate should be excluded from the gain calculation.
